SigAverager/testing/test_dsp_signal_averager.py

```diff
@@ -0,0 +1,128 @@
+"""Unit test for signal averager."""
+import SigAverager
+import numpy as np
+import logging
+import matplotlib.pyplot as plt
+
+
+def test_signal_averaging():
+    """
+    Test signal averaging.
+
+    Test Overview:
+    --------------
+    This test will pass parametets to signal averaging method and get back a vector.
+    The intention is to check for the correct averaging. The SNR should improve with the number of iterations.
+    The test CW tone is 75MHz (EE3:75).
+    The test will look for the following:
+    a) Is the SNR correct? This is verified using an FFT and computing the SNR.
+
+    Parameters
+    ----------
+    cw_scale: float
+        factor to scale generated noise.
+    cw_freq: float
+        Frequency of CW to be generated.
+    sampling_frequency: int
+        Sample rate for generated CW. This is expressed in Hz. E.g. 1712e6.
+    num_samples: int
+        Number of samples for generated CW.
+    noise_scale: float
+        Factor to scale generated noise.
+    n_iter: int
+        Number of iterations to average signal.
+    expected_SNR_dB: int
+        Expected SNR after signal averaging.
+    """
+    cw_scale = 0.01
+    cw_freq = 75e6
+    sampling_frequency = 1712e6
+    num_samples = 8192
+    noise_scale = 0.4
+    n_iter = 2048
+    expected_SNR_dB = 50
+
+    # Compute Averaged Signal
+    averaged_signal = SigAverager.signal_averager(
+        cw_scale, cw_freq, sampling_frequency, noise_scale, num_samples, n_iter
+    )
+
+    # Compute real FFT on average signal
+    ave_sig_fft = np.power(np.fft.rfft(np.real(averaged_signal), axis=-1), 2)
+    ave_sig_fft_channel = np.where(ave_sig_fft == np.max(ave_sig_fft))[0][0]
+
+    tone_max = max(np.abs(ave_sig_fft))
+    ave_sig_max_tone_removed = ave_sig_fft
+    ave_sig_max_tone_removed[ave_sig_fft_channel] = 0
+    ave_sig_mean = 0
+    for idx in range(len(ave_sig_max_tone_removed)):
+        ave_sig_mean += ave_sig_max_tone_removed[idx]
+    ave_sig_mean = np.abs(ave_sig_mean) / len(ave_sig_max_tone_removed)
+
+    ave_sig_SNR = 10 * np.log10(tone_max / ave_sig_mean)
+
+    # Plot it illustrate (if required)
+    print(f"SNR is: {ave_sig_SNR}")
+    plt.figure(1)
+    plt.plot(averaged_signal[0:256])
+
+    plt.figure(2)
+    plt.plot(np.abs(ave_sig_fft))
```

Comment:
for the X-axis, consider using `np.fft.rfftfreq()`.

Reply:
It is used on line 50.